### PR TITLE
Scippified Predict posterior

### PR DESCRIPTION
**Pull request by Harry-Rich**

### CODE REVIEW

File: kinisi/diffusion.py [hunk 1]
```diff
@@ -194,7 +195,7 @@ def compute_covariance_matrix(self) -> sc.Variable:
         :returns: A :py:mod:`scipp` object containing the covariance matrix.
         """
         cov = np.zeros((self.msd.data.variances.size, self.msd.data.variances.size))
-        for i in range(0, self.msd.data.variances.size):
+        for i in tqdm(range(0, self.msd.data.variances.size)):
```
Comment (arm61) [inline kinisi/diffusion.py:198]: Is this cause the number of data points can be really large? I would prefer if we include it, we do it at a later date.

File: kinisi/diffusion.py [hunk 2]
```diff
@@ -205,6 +206,53 @@ def compute_covariance_matrix(self) -> sc.Variable:
...
+            shape=[n_posterior_samples, n_predictive_samples, self.msd.coords['time interval'][diff_regime:].size],
+            unit=ppd_unit)
+        samples_to_draw = list(enumerate(np.random.randint(0, self.gradient.size, size=n_posterior_samples)))
+
+        if progress:
+            iterator = tqdm(samples_to_draw, desc='Calculating Posterior Predictive')
+        else:
+            iterator = samples_to_draw
+
+        # Testing unit consistency for mu and covariance before producing mv
```
Comment (arm61) [inline kinisi/diffusion.py:242]:
```suggestion
        # Checking unit consistency for mu and covariance
```

File: kinisi/diffusion.py [hunk 3]
```diff
@@ -205,6 +206,53 @@ def compute_covariance_matrix(self) -> sc.Variable:
...
+        try:
+            ppd_unit**2 == self._covariance_matrix.unit
```
Comment (arm61) [inline kinisi/diffusion.py:244]: Is this ever not true?

File: kinisi/diffusion.py (hunk 1 above)
Comment (Harry-Rich) [reply kinisi/diffusion.py:198]: Ah think this was from me messing around testing, forgot to pull from repo before starting on predict_posterior

File: kinisi/diffusion.py (hunk 3 above)
Comment (Harry-Rich) [reply kinisi/diffusion.py:244]: Wasn't sure, just thought better safe than sorry as we use the values directly in the multivariate normal

**Pull request merged**